pkg/tf2pulumi/convert/tf.go

```diff
@@ -1281,8 +1281,50 @@ func convertBody(sources map[string][]byte, scopes *scopes, fullyQualifiedPath s
 	// If we see blocks we turn those into lists (unless maxItems==1)
 	blockLists := make(map[string][]bodyAttrsTokens)
 	for _, block := range content.Blocks {
-		// TODO: We need to correctly handle dynamic blocks somehow.
 		if block.Type == "dynamic" {
+			eachVar := scopes.generateUniqueName("entry", "", "")
+			dynamicTokens := hclwrite.Tokens{makeToken(hclsyntax.TokenOBrack, "[")}
+			dynamicTokens = append(dynamicTokens, makeToken(hclsyntax.TokenIdent, "for"))
+			dynamicTokens = append(dynamicTokens, makeToken(hclsyntax.TokenIdent, eachVar))
+			dynamicTokens = append(dynamicTokens, makeToken(hclsyntax.TokenIdent, "in"))
+			dynamicBody, ok := block.Body.(*hclsyntax.Body)
+			if !ok {
+				continue
+			}
+
+			forEachAttr, hasForEachAttr := dynamicBody.Attributes["for_each"]
+			if !hasForEachAttr {
+				continue
```

Comment:
Is it valid to have a dynamic block without for_each? Should it just set the attribute to `[]`?